pycbc/waveform/compress.py

```diff
@@ -257,7 +257,16 @@ def compress_waveform(htilde, sample_points, tolerance, interpolation,
         minpt = vecdiffs.argmax()
         # add a point at the frequency halfway between minpt and minpt+1
         add_freq = sample_points[[minpt, minpt+1]].mean()
-        addidx = int(add_freq/df)
+        addidx = int(round(add_freq/df))
+        # ensure that only new points are added
+        while addidx in sample_index:
+            addpt -= 1
```

Comment:
@cdcapano Landscape is complaining about addpt not being defined before. Should it be set to a value -1 somewhere before the while loop?